client/src/domain/Community/Community.js

```diff
@@ -0,0 +1,89 @@
+import "./Community.css";
+import { Link } from "react-router-dom";
+
+import CommunityFallbackProfilePic from "../../assets/happy-students.jpg";
+
+import { FaRegEdit } from "react-icons/fa";
+import { FaTrashAlt } from "react-icons/fa";
+import axios from "axios";
+
+import CommunitiesContext from "../../contexts/CommunitiesContext";
+import { useContext } from "react";
+
+import { useHistory } from "react-router";
+
+// TODO: Get rid of that conditional wrapper again - shoudl be possible to just set an onLick-handler on the section-tag itself and basrta kanzla!
+const ConditionalWrapper = ({ condition, wrapper, children }) =>
+  condition ? wrapper(children) : children;
```

Comment:
Can you explain this to me?

Reply:
Of course - this is a frackin awesome construct ( a higher order component, if I'm not mistaken) that allows me to conditionally wrap a component inside an anchor tag - I found that here:
https://blog.hackages.io/conditionally-wrap-an-element-in-react-a8b9a47fab2

It implicitly accepts code blocks (here jsx) as last argument - like in godd ol Ruby blocks (I'm refering to the last parameter here - "children" )

(same source): "...the children prop is actually a React feature. Each component receives an implicit prop called children that contains all the elements contained between the opening tag and the closing tag where it is called."

That's more or less similar to Ruby Blocks, which can be passed as implicit last argument to a method... I thought that wasn't possible in JS - but it is, yay - which opens up a couple of interestng possibilities...